### PR TITLE
More validation for rag strategies

### PR DESCRIPTION
**Pull request by lferran**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/nucliadb/search/search/chat/prompt.py [hunk 1]
```diff
@@ -210,7 +210,11 @@ async def get_resource_field_extracted_text(
     if resource is None:
         return None
 
-    field_type, field_key = field_id.split("/")
+    try:
+        field_type, field_key = field_id.strip("/").split("/")
```
Comment (javitonino) [inline nucliadb/nucliadb/search/search/chat/prompt.py:214]: We added `.strip("/")` here to remove leading/trailing `/`. Is it really needed? It looks like it makes the validation less strict, which we may want or not.

If this is the case we may want to have a test to ensure leading/trailing `/` are ignored.

File: nucliadb_models/nucliadb_models/search.py [hunk 2]
```diff
@@ -737,15 +737,53 @@ class RagStrategy(BaseModel):
...
 
+    @root_validator(pre=True)
+    def fields_validator(cls, values):
+        if values.get("fields") is None:
+            return values
+
+        # Check that the fields are in the format {field_type}/{field_name}
+        for field in values.get("fields"):
+            try:
+                field_type, _ = field.strip("/").split("/")
```
Comment (javitonino) [inline nucliadb_models/nucliadb_models/search.py:769]: Same as above

File: nucliadb_models/nucliadb_models/search.py [hunk 3]
```diff
@@ -737,15 +737,53 @@ class RagStrategy(BaseModel):
...
+            except ValueError:
+                raise ValueError(
+                    f"Field '{field}' is not in the format {{field_type}}/{{field_name}}"
+                )
+            if field_type not in ALLOWED_FIELD_TYPES:
+                allowed_field_types_part = ", ".join(
+                    [
+                        f"'{fid}' for '{fname}' fields"
```
Comment (javitonino) [inline nucliadb_models/nucliadb_models/search.py:777]: Nice detailed error message :sparkling_heart:

**Review by javitonino — APPROVED** [review]
Left a question, not sure if `.strip()` is right. If it is, maybe add a test? And then :rocket:

File: nucliadb/nucliadb/search/search/chat/prompt.py (hunk 1 above)
Comment (lferran) [reply nucliadb/nucliadb/search/search/chat/prompt.py:214]: Yes! I added the strip so that trailing/leading / were ignored. Will add the test!!

**Pull request merged**